BUILD.gn

```diff
@@ -480,6 +480,7 @@ source_set("electron_lib") {
       "SecurityInterface.framework",
       "ServiceManagement.framework",
       "StoreKit.framework",
+      "UserNotifications.framework",
```

Comment:
We can't do this, afaik just _linking_ to this framework breaks some usages of the `NSUserNotification` API.  We would need a complete migration to the new framework which is a larger amount of work we don't want to do right now

Reply:
Do you know which ones?

Reply:
This is important for us, as we have a custom node module using the `UserNotifications.framework` and we are not getting the notification information when the app is launched from the notification. I am willing to spend time on the migration if that's necessary.

Reply:
@miniak I'm not 100% which ones, but to be safe we should (a) test it and figure it out or (b) start the process of using `@available` declarations and use the new `UN` APIs on 10.14 and above where they are available.

Reply:
@miniak I've thought about this a bit and you should actually be able to implement this entire thing using fake declarations, `respondsToSelector` and static casts without ever actually linking to the `UN` framework

Reply:
btw how do you know that just linking `UserNotifications.framework` will caused `NSUserNotification` based  notifications to stop working?

Reply:
@MarshallOfSound I've changed the code to load the framework dynamically, is that enough? Or do we have to avoid it completely?

Reply:
> btw how do you know that just linking UserNotifications.framework will cause NSUserNotification based notifications to stop working?

Horror stories from a while ago, I can't find the reference right now.

> I've changed the code to load the framework dynamically, is that enough? Or do we have to avoid it completely?

I think you have to avoid the symbols completely 🤔

Reply:
I don't load the framework anymore, symbols are only needed to get the class for instance check, which I've replaced  by checking for selectors.